backend/de.metas.deliveryplanning.base/src/main/java/de/metas/deliveryplanning/process/M_Delivery_Planning_GenerateShortageOverage.java

```diff
@@ -0,0 +1,334 @@
+
+		return inventoryRecord;
+	}
+
+	private void createInventoryLine(@NonNull final I_M_Inventory inventoryRecord, @NonNull final I_M_InOut inOutRecord)
+	{
+		final InventoryId inventoryId = InventoryId.ofRepoId(inventoryRecord.getM_Inventory_ID());
+		final List<I_M_InOutLine> inOutLines = inOutDAO.retrieveLines(inOutRecord);
+		final I_M_InOutLine firstInOutLineRecord = inOutLines.get(0);
+		final LocatorId locatorId = warehouseDAO.getLocatorIdByRepoIdOrNull(firstInOutLineRecord.getM_Locator_ID());
+		Check.assumeNotNull(locatorId, "locator shall be set");
+
+		final ProductId productId = ProductId.ofRepoId(firstInOutLineRecord.getM_Product_ID());
+		final I_C_UOM stockingUOM = productBL.getStockUOM(productId);
+
+		final AttributeSetInstanceId asiId = AttributeSetInstanceId.ofRepoIdOrNone(firstInOutLineRecord.getM_AttributeSetInstance_ID());
+		final WarehouseId warehouseId = WarehouseId.ofRepoId(inventoryRecord.getM_Warehouse_ID());
+
+		final TableRecordReference mInOutLineTableRecordReference = TableRecordReference.of(firstInOutLineRecord);
+		final Set<TableRecordReference> mInOutLineTableRecordReferenceSet = Collections.singleton(mInOutLineTableRecordReference);
+		final ImmutableSetMultimap<TableRecordReference, HuId> huIdsMultimap = huAssignmentDAO.retrieveHUsByRecordRefs(mInOutLineTableRecordReferenceSet);
+		final Collection<HuId> huIds = huIdsMultimap.get(mInOutLineTableRecordReference);
+
+		final ShortageAndOverageStrategy shortageAndOverageStrategy = HUsForInventoryStrategies.shortageAndOverage()
+				.warehouseId(warehouseId)
+				.locatorId(locatorId)
+				.productId(productId)
+				.huForInventoryLineFactory(huForInventoryLineFactory)
+				.huIds(huIds)
+				.build();
+		final List<HuForInventoryLine> hus = shortageAndOverageStrategy.streamHus()
+				.collect(ImmutableList.toImmutableList());
+
+		final Quantity qtyBooked = computeTotalQtyBooked(hus, stockingUOM, UOMConversionContext.of(productId));
+		if(qtyBooked.toBigDecimal().add(p_Qty).signum() < 0)
+		{
+			throw new AdempiereException("Shortage is greater than booked quantity");
+		}
+
+		final I_M_InventoryLine inventoryLineRecord = InterfaceWrapperHelper.newInstance(I_M_InventoryLine.class);
+		inventoryLineRecord.setM_Inventory_ID(inventoryRecord.getM_Inventory_ID());
+		inventoryLineRecord.setAD_Org_ID(inventoryRecord.getAD_Org_ID());
+		inventoryLineRecord.setM_Locator_ID(locatorId.getRepoId());
+		inventoryLineRecord.setM_Product_ID(productId.getRepoId());
+		inventoryLineRecord.setM_AttributeSetInstance_ID(asiId.getRepoId());
+		inventoryLineRecord.setHUAggregationType(HUAggregationType.toCodeOrNull(HUAggregationType.SINGLE_HU));
+		inventoryLineRecord.setC_UOM_ID(qtyBooked.getUomId().getRepoId());
+		inventoryLineRecord.setQtyCount(qtyBooked.toBigDecimal().add(p_Qty));
+		inventoryLineRecord.setQtyBook(qtyBooked.toBigDecimal());
+		inventoryLineRecord.setIsCounted(true);
+		inventoryLineRecord.setM_HU_ID(huIds.stream().findFirst().get().getRepoId());
```

Comment:
- better use `hus` maybe you don'T even need a dedicated variable for `huIds`
- use `CollectionUtils#singleElement(java.util.Collection<T>)` to get he single item and assert that there is just one

Reply:
Later edit: as talked, please check out `DraftInventoryBase` and its existing subclasses, and create a new one. if that doesn't work, let's talk about why it doesn't